pkg/client/cache/index.go

```diff
@@ -14,10 +16,41 @@ const (
 // ImageStreamReferenceIndexFunc is a default index function that indexes based on image stream references.
 func ImageStreamReferenceIndexFunc(obj interface{}) ([]string, error) {
 	switch t := obj.(type) {
-	// TODO: Add support for build configs
-	case *deployapi.DeploymentConfig:
+	case *buildapi.BuildConfig:
 		var keys []string
+		for _, trigger := range t.Spec.Triggers {
+			if trigger.Type != buildapi.ImageChangeBuildTriggerType {
+				continue
+			}
+			from := trigger.ImageChange.From
+
+			// the default imagechange trigger has no "from" value, it
+			// uses the input image of the buildconfig as the imagestream
+			// to trigger off, so we need to look that up.
+			if from == nil {
+				from = buildutil.GetInputReference(t.Spec.Strategy)
+				if from == nil || from.Kind != "ImageStreamTag" {
+					continue
+				}
+			}
+			name, _, _ := imageapi.SplitImageStreamTag(from.Name)
+			namespace := from.Namespace
+
+			// if the imagestream reference has no namespace, use the
+			// namespace of the buildconfig.
+			if len(namespace) == 0 {
+				namespace = t.Namespace
+			}
+			keys = append(keys, namespace+"/"+name)
+		}
 
+		if len(keys) == 0 {
```

Comment:
I copied this from the deployments implementation a few lines down...  @kargakis ?